src/mlpack/core/data/save_impl.hpp

```diff
@@ -283,6 +283,126 @@ bool Save(const std::string& filename,
   }
 }
 
+#ifdef HAS_STB
+// Image saving API.
+template<typename eT>
+bool Save(const std::string& filename,
+          arma::Mat<eT>& matrix,
+          ImageInfo& info,
+          const bool fatal,
+          const bool transpose)
+{
+  Timer::Start("saving_image");
+  // We transpose by default. So, un-transpose if necessary.
+  if (!transpose)
+    matrix = arma::trans(matrix);
+
+  int tempWidth, tempHeight, tempChannels, tempQuality;
+
+  tempWidth = info.Width();
+  tempHeight = info.Height();
+  tempChannels = info.Channels();
+  tempQuality = info.Quality();
+
+  if (!ImageFormatSupported(filename, true))
+  {
+    std::ostringstream oss;
+    oss << "File type " << Extension(filename) << " not supported.\n";
+    oss << "Currently it supports ";
+    for (auto extension : saveFileTypes)
+      oss << ", " << extension;
+    oss << std::endl;
+    throw std::runtime_error(oss.str());
+    return false;
+  }
+  if (matrix.n_cols > 1)
+  {
+    std::cout << "Input Matrix contains more than 1 image." << std::endl;
+    std::cout << "Only the firstimage will be saved!" << std::endl;
+  }
+  stbi_flip_vertically_on_write(transpose);
+
+  bool status = false;
+  try
+  {
+    unsigned char* image = matrix.memptr();
+
+    if ("png" == Extension(filename))
+    {
+      status = stbi_write_png(filename.c_str(), tempWidth, tempHeight,
+          tempChannels, image, tempWidth * tempChannels);
+    }
+    else if ("bmp" == Extension(filename))
+    {
+      status = stbi_write_bmp(filename.c_str(), tempWidth, tempHeight,
+          tempChannels, image);
+    }
+    else if ("tga" == Extension(filename))
+    {
+      status = stbi_write_tga(filename.c_str(), tempWidth, tempHeight,
+          tempChannels, image);
+    }
+    else if ("hdr" == Extension(filename))
+    {
+      status = stbi_write_hdr(filename.c_str(), tempWidth, tempHeight,
+          tempChannels, reinterpret_cast<float*>(image));
+    }
+    else if ("jpg" == Extension(filename))
+    {
+      status = stbi_write_jpg(filename.c_str(), tempWidth, tempHeight,
+          tempChannels, image, tempQuality);
+    }
+  }
+  catch (std::exception& e)
+  {
+    Timer::Stop("saving_image");
+    if (fatal)
+      Log::Fatal << e.what() << std::endl;
+    else
```

Comment:
We can remove the else here, ``Fatal``will exit immediately, so:

```c++
Timer::Stop("saving_image");
if (fatal)
  Log::Fatal << e.what() << std::endl;

Log::Warn << e.what() << std::endl;
return false;
```

will work as well.

Reply:
I did add it because it was it the rest of the code.

Reply:
Agreed, I think I wrote a lot of the offending code with the `if` and `else`, but the `else` isn't needed.  If I see it in the future I'll change it and open a PR. :)